// minify-minifiers/index.js
var t = require('minify-util').t;
var shell = require('shelljs');
var requirejs = require('requirejs');

exports.css = function (input, output, app) {
	requirejs.optimize({
		cssIn       : input,
		out         : output,
		optimizeCss : 'standard'
	});
};

exports.js = function (input, output, app) {
	var uglify = require('uglify-js2');
	var result = uglify.minify(input);
	result.code.to(output);
};

exports.php = function (input, output, app) {
	var cmd = 'java -jar {{dir}}/bin/htmlcompressor-1.5.3.jar {{input}} -o {{output}} --preserve-php';
	var args = {
		dir    : app.__dirname,
		input  : input,
		output : output
	};
	shell.exec(t(cmd, args), {silent: true});
};

exports.png = function (input, output, app) {
	var cmd = 'pngcrush -rem alla -reduce -brute {{input}} {{output}}';
	var args = {
		input  : input,
		output : output
	};
	shell.exec(t(cmd, args), {silent: true});
};

exports.jpg = function (input, output, app) {
	var cmd = 'jpegtran -copy none -optimize -perfect -outfile {{output}} {{input}}';
	var args = {
		input  : input,
		output : output
	};
	shell.exec(t(cmd, args), {silent: true});
};

exports.html = exports.php;
exports.jpeg = exports.jpg;
